### PR TITLE
Implement composition UI in Twitter

### PR DESCRIPTION
**Pull request by guanbinrui**
This PR includes:

+ Add `PostDialog` as original `AdditionalPostBox`;
+ Add `useTheme` replace `useColorScheme `;
+ Update tweet text payload template;
+ Use Twitter theme color as primary color of material UI;

Related issues:

+ Post Composition UI - Temporary Consolidation (#397 )
+ [Demand] Implement Twitter payload (RFC 373) (#456)

Screenshots:

image
image
image

### CODE REVIEW

**Comment by neruthes** [comment]
Looks good

File: src/components/shared/SelectRecipients/GroupInChip.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            if (props.onChange) {
+                props.onChange(ev, !props.checked)
+            }
+        },
+        [props],
+    )
+
+    return (
+        <Chip
+            className={classes.root}
```
Comment (Jack-Works) [inline src/components/shared/SelectRecipients/GroupInChip.tsx]: this way will lost the classes.root, maybe you should write {...props} className={classnames(classes.root, props.className)}

File: src/components/shared/SelectRecipients/ProfileInList.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+export function ProfileInList(props: ProfileInListProps) {
+    const classes = useStylesExtends(useStyle(), props)
+    const profile = props.item
+    const name = profile.nickname || profile.identifier.userId
+    const secondary = profile.linkedPersona?.fingerprint.toLowerCase()
+    const onClick = useCallback(ev => props.onChange(ev, !props.checked), [props])
+
+    return (
+        <ListItem className={classes.root} button disabled={props.disabled} onClick={onClick} {...props.ListItemProps}>
```
Comment (Jack-Works) [inline src/components/shared/SelectRecipients/ProfileInList.tsx]: this way will lost the classes.root, maybe you should write {...props} className={classnames(classes.root, props.className)}

File: src/components/shared/SelectRecipients/SelectRecipientsDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import { ProfileInList } from './ProfileInList'
+import { Profile } from '../../../database'
+import { useCurrentIdentity } from '../../DataSource/useActivatedUI'
+import { ProfileIdentifier } from '../../../database/type'
+
+const useStyles = makeStyles(theme => ({
+    dialog: {},
+    backdrop: {},
+    paper: {},
+    header: {},
```
Comment (Jack-Works) [inline src/components/shared/SelectRecipients/SelectRecipientsDialog.tsx]: if you're not using those empty classes, you can add it to the withClasses declaration above.

File: src/components/Welcomes/Banner.tsx [hunk 4]
```diff
@@ -109,26 +114,30 @@ export function BannerUI(props: Props) {
...
-            <Button onClick={props.close.onClose} color="primary">
+            <Button className={classes.button} color="primary" onClick={props.close.onClose}>
                 {geti18nString('cancel')}
             </Button>
         ) : null
     return (
         <AppBar position="static" color="inherit" elevation={0} classes={{ root: classes.root }}>
-            <DialogTitle classes={{ root: classes.title }}>{Title}</DialogTitle>
-            <DialogContent>
+            <DialogTitle classes={{ root: classes.title }} className={classes.header}>
```
[the remark this answers is not in the document]
Comment (guanbinrui) [reply src/components/Welcomes/Banner.tsx]: Yes both class names will keep.

File: src/components/InjectedComponents/PostDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+        MessageCenter.on('compositionUpdated', onChange)
+        return () => {
+            MessageCenter.off('compositionUpdated', onChange)
+        }
+    }, [identities.length, props.reason])
+
+    const [postBoxText, setPostBoxText] = useState('')
+    const [currentShareTarget, setCurrentShareTarget] = useState(availableShareTarget)
+    const onFinishButtonClicked = useCallback(() => {
+        onRequestPost(currentShareTarget, postBoxText)
```
[the remark this answers is not in the document]
Comment (guanbinrui) [reply src/components/InjectedComponents/PostDialog.tsx]: It's a copy from <AdditionalPostBox /> and union it with onRequestReset() in order to reduce rerender times.

File: src/components/InjectedComponents/PostDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+                            warningText: geti18nString('additional_post_box__encrypted_failed'),
+                            shouldOpenPostDialog: false,
+                        },
+                    )
+                    activeUI.taskUploadToPostBox(encrypted, {
+                        warningText: geti18nString('additional_post_box__steganography_post_failed'),
+                    })
+                } else {
+                    activeUI.taskPasteIntoPostBox(
+                        geti18nString(props.textPayloadKey ?? 'additional_post_box__encrypted_post_pre', encrypted),
```
Comment (guanbinrui) [inline src/components/InjectedComponents/PostDialog.tsx]: Hi @Jack-Works @SunriseFox Here twitter and facebook use different template in same message record. So may be we should update the message structure in `message.json` like this:

```json
{
    "additional_post_box__encrypted_post_pre": {
        "facebook.com": {
            "message": "Decrypt this post with #Maskbook ! $1",
            "description": "$1 should at the end of the string."
        },
        "twitter.com": {
            "message": "#Maskbook ([I:b])\nDecrypt this tweet with maskbook.com @ProjectMaskbook.\n—§— /* {$1} */"
        }
    }
}
```

What do you think about it?

Comment (Jack-Works) [reply src/components/InjectedComponents/PostDialog.tsx]: @guanbinrui this format is originally comes from https://developer.chrome.com/extensions/i18n and we may go back to old format so it's better not to change the format other than the chrome one

Comment (Jack-Works) [reply src/components/InjectedComponents/PostDialog.tsx]: I'll add a commit to this pull request resolve this problem.

Comment (Jack-Works) [reply src/components/InjectedComponents/PostDialog.tsx]: @guanbinrui with https://github.com/DimensionDev/Maskbook/pull/499/commits/8e8682b9923a3e2a862698d9d9e4c69ceb2e1b81 I think you can resolve this problem now

Comment (guanbinrui) [reply src/components/InjectedComponents/PostDialog.tsx]: cool~ looks good!

Comment (guanbinrui) [reply src/components/InjectedComponents/PostDialog.tsx]: Hi @Jack-Works I refactored it with `8e1ca8079f77761defc1ed6c5d5e37fa14e42dfb`. Pls have a look!

**Comment by guanbinrui** [comment]
See #518

**Pull request closed without merging**